### DISCUSSION TITLE
Vnt Desgin React uses automatic on-demand loading

### DISCUSSION THREAD

**Discussion by XiaoYuTongXue-ciy** [ant-design/ant-design#51488] [Q&A]
After reading the latest documents of Vnt Desgin Vue and Vnt Desgin React, I found that there is a method for automatic on-demand loading for Vnt Desgin Vue, but there is no automatic on-demand loading function in the latest version of Vnt Desgin React. I was able to use it for React projects a year ago. If my understanding is wrong, please tell me the correct way to use it

**Comment by petercat-assistant** [comment]
I couldn't find any specific discussions or issues directly related to the automatic on-demand loading feature for Vnt Design React in the repository. It's possible that the feature might have been deprecated or changed in the latest version.

For automatic on-demand loading in React, you can typically use tools like `babel-plugin-import` which allows you to load components on demand. Here's a basic example of how you might set it up:

1. Install the plugin:
   ```bash
   npm install babel-plugin-import --save-dev
   ```

2. Configure your Babel configuration (e.g., `.babelrc`):
   ```json
   {
     "plugins": [
       ["import", { "libraryName": "vnt-design", "libraryDirectory": "es", "style": "css" }]
     ]
   }
   ```

This setup allows you to import components from Vnt Design React as needed, reducing the bundle size by only including the components you use.

If this doesn't resolve your issue or if you have further questions, please feel free to ask!

For further assistance, please describe your question in the comments and @petercat-assistant to start a conversation with me.

**Reply by XiaoYuTongXue-ciy** [reply]
This method doesn't help me solve this problem。
